pkg/bpf/collection_test.go

```diff
@@ -117,9 +120,19 @@ func TestInlineGlobalData(t *testing.T) {
 		t.Errorf("unexpected Instruction constant: want: 0x%x, got: 0x%x", want, got)
 	}
 
-	if want, got := 0x7f00000000000000, int(insns[2].Constant); want != got {
+	if want, got := 42, int(insns[2].Constant); want != got {
 		t.Errorf("unexpected Instruction constant: want: 0x%x, got: 0x%x", want, got)
 	}
+
+	spec = orig.Copy()
+	if err := inlineGlobalData(spec, map[string]uint64{"bogus": 42}); err == nil {
+		t.Fatal("Expected error when trying to replace bogus constant")
+	}
+
+	delete(spec.Maps, globalDataMap)
+	if err := inlineGlobalData(spec, map[string]uint64{"bogus": 42}); err != nil {
```

Comment:
Should we leave this nil to isolate the error condition better?
```suggestion
	if err := inlineGlobalData(spec, nil}); err != nil {
```